### PR TITLE
PHP: Added support for Psalm static analyzer

### PR DESCRIPTION
**Pull request by troizet**
This PR adds support for the `Psalm` static analyzer.
https://psalm.dev/

https://github.com/user-attachments/assets/2428ca68-a920-4399-8138-b8af43b7fb10

Most of it is copying code from `PHPStan`. The main differences are in the command to start the analyzer in [Psalm.java.](https://github.com/troizet/netbeans/blob/4621fee2683e91cb55985e0ba4993d32910681f8/php/php.code.analysis/src/org/netbeans/modules/php/analysis/commands/Psalm.java)

Also in this PR the file `PHPStanStyleReportParser.java` is renamed to `CheckStyleReportParser.java` and is used for both `PHPStan` and `Psalm`.

Also, the image file for `Psalm` is a copy of `textDetail.png` from the `Search API` [module](https://github.com/apache/netbeans/blob/534340c73115d93815fcd0017373e150ac470279/platform/api.search/src/org/netbeans/modules/search/res/textDetail.png).

Tested on `Psalm` version 5.26.1

### CODE REVIEW

File: php/php.code.analysis/src/org/netbeans/modules/php/analysis/ImportantFilesImpl.java [hunk 1]
```diff
@@ -42,7 +43,9 @@ public final class ImportantFilesImpl implements ImportantFilesImplementation {
             CodingStandardsFixer.DIST_CONFIG_FILE_NAME_V3,
             PHPStan.CONFIG_FILE_NAME,
             PHPStan.DIST_CONFIG_FILE_NAME,
-            PHPStan.ALTERNATIVE_DIST_CONFIG_FILE_NAME};
+            PHPStan.DIST_CONFIG_FILE_NAME,
```
Comment (pincher2012) [inline php/php.code.analysis/src/org/netbeans/modules/php/analysis/ImportantFilesImpl.java:46]: Two identical constants here

Comment (troizet) [reply php/php.code.analysis/src/org/netbeans/modules/php/analysis/ImportantFilesImpl.java]: Fixed. Thank you!

**Review by tmysik — COMMENTED** [review]
Looks good to me; please, let @junichi11 review it too. Thanks!

File: php/php.code.analysis/src/org/netbeans/modules/php/analysis/parsers/CheckStyleReportParser.java [hunk 2]
```diff
@@ -48,10 +48,10 @@
 /**
  * Parser for PHPStan xml report file.
  */
-public class PHPStanReportParser extends DefaultHandler {
+public class CheckStyleReportParser extends DefaultHandler {
```
[the remark this answers is not in the document]
Comment (tmysik) [reply php/php.code.analysis/src/org/netbeans/modules/php/analysis/parsers/CheckStyleReportParser.java:51]: I mean, is it generic enough to be shared between more static analyzers?

[the remark this answers is not in the document]
Comment (troizet) [reply php/php.code.analysis/src/org/netbeans/modules/php/analysis/parsers/CheckStyleReportParser.java:51]: Yes, the `checkstyle` output format is supported by most static analyzers: 
- [PHPStan](https://phpstan.org/user-guide/output-format), 
- [PHP Coding Standards Fixer](https://cs.symfony.com/doc/usage.html), 
- [PHP_CodeSniffer](https://github.com/PHPCSStandards/PHP_CodeSniffer/wiki/Usage),
- PSalm

[the remark this answers is not in the document]
Comment (tmysik) [reply php/php.code.analysis/src/org/netbeans/modules/php/analysis/parsers/CheckStyleReportParser.java:51]: Perfect, thanks for the clarification.

File: php/php.code.analysis/src/org/netbeans/modules/php/analysis/commands/Psalm.java [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+
+    private ExecutionDescriptor getDescriptor() {
+        return PhpExecutable.DEFAULT_EXECUTION_DESCRIPTOR
+                .optionsPath(AnalysisOptionsPanelController.OPTIONS_PATH)
+                .frontWindowOnError(false)
+                .inputVisible(false)
+                .preExecution(() -> {
+                    if (XML_LOG.isFile()) {
+                        if (!XML_LOG.delete()) {
+                            LOGGER.log(Level.INFO, "Cannot delete log file {0}", XML_LOG.getAbsolutePath());
```
Comment (junichi11) [inline php/php.code.analysis/src/org/netbeans/modules/php/analysis/commands/Psalm.java:176]: `// NOI18N`

File: php/php.code.analysis/src/org/netbeans/modules/php/analysis/parsers/CheckStyleReportParser.java [hunk 4]
```diff
@@ -48,10 +48,10 @@
 /**
  * Parser for PHPStan xml report file.
```
Comment (junichi11) [inline php/php.code.analysis/src/org/netbeans/modules/php/analysis/parsers/CheckStyleReportParser.java:49]:
```suggestion
 * Parser for PHPStan and Psalm xml report files.
```

**Review by junichi11 — COMMENTED** [review]
It's great because this is implemented based on the existing code :)
Just in case, I would add unit tests using the actual results of an example. Thanks!

File: php/php.code.analysis/src/org/netbeans/modules/php/analysis/commands/Psalm.java (hunk 3 above)
Comment (troizet) [reply php/php.code.analysis/src/org/netbeans/modules/php/analysis/commands/Psalm.java]: Done.

File: php/php.code.analysis/src/org/netbeans/modules/php/analysis/parsers/CheckStyleReportParser.java (hunk 4 above)
Comment (troizet) [reply php/php.code.analysis/src/org/netbeans/modules/php/analysis/parsers/CheckStyleReportParser.java]: Done.

**Comment by troizet** [comment]
> I would add unit tests using the actual results of an example

Added a test for parsing psalm output.

**Review by tmysik — APPROVED** [review]

File: php/php.code.analysis/test/unit/data/psalm/PsalmSupport/test/src/CalculatorTest.php [hunk 5]
```diff
@@ -0,0 +1,3 @@
+<?php
+/* Licensed to the Apache Software Foundation (ASF) under one or more contributor license agreements; and to You under the Apache License, Version 2.0. */
+// dummy file
```
Comment (junichi11) [inline php/php.code.analysis/test/unit/data/psalm/PsalmSupport/test/src/CalculatorTest.php:3]: Nitpick: add a new line

**Review by junichi11 — APPROVED** [review]
Looks good. Thank you!

File: php/php.code.analysis/test/unit/data/psalm/PsalmSupport/test/src/CalculatorTest.php (hunk 5 above)
Comment (troizet) [reply php/php.code.analysis/test/unit/data/psalm/PsalmSupport/test/src/CalculatorTest.php]: Done.

**Pull request merged**